tests/integration/test_active_response/test_execd/test_execd_restart.py

```diff
@@ -0,0 +1,168 @@
+    authd_simulator.shutdown()
+
+@pytest.fixture(scope="module", params=configurations)
+def get_configuration(request):
+    """Get configurations from the module"""
+    yield request.param
+
+def wait_message_line(line):
+    if platform.system() == 'Windows' and "active-response/bin/restart-wazuh.exe: {\"version\"" in line:
+        return True
+    elif ("ossec/active-response/bin/restart-wazuh: {\"version\"" in line):
+        return True
+    return None
+
+def wait_invalid_input_message_line(line):
+    if ("Invalid input format" in line):
+        return line
+    return None
+
+def wait_shutdown_message_line(line):
+    if ("Shutdown received. Deleting responses." in line):
+        return True
+    return None
+
+def build_message(metadata, expected):
+    origin = "\"name\":\"\",\"module\":\"wazuh-analysisd\""
+    rules = "\"level\":5,\"description\":\"Test.\",\"id\":" + metadata['rule_id']
+
+    if expected['success'] == False:
+        return "{\"origin\":{" + origin + "},\"command\":\"" + metadata['command'] + "\",\"parameters\":{\"extra_args\":[],\"alert\":{\"rule\":{" + rules + "}}}}"
+
+    return "{\"version\":1,\"origin\":{" + origin + "},\"command\":\"" + metadata['command'] + "\",\"parameters\":{\"extra_args\":[],\"alert\":{\"rule\":{" + rules + "}}}}"
+
+def test_execd_restart(set_debug_mode, get_configuration, test_version, configure_environment, start_agent, set_ar_conf_mode):
+    metadata = get_configuration['metadata']
+    expected = metadata['results']
+    ossec_log_monitor = FileMonitor(LOG_FILE_PATH)
+    ar_log_monitor = FileMonitor(AR_LOG_FILE_PATH)
+
+    ##### Checking AR in ossec logs ####
+    try:
+        ossec_log_monitor.start(timeout=60, callback=wait_received_message_line)
+    except TimeoutError as err:
+        raise AssertionError("Received message tooks too much!")
+
+    ##### Checking AR in active-response logs ####
+    try:
+        ar_log_monitor.start(timeout=60, callback=wait_start_message_line)
+    except TimeoutError as err:
+        raise AssertionError("Start message tooks too much!")
+
+    if expected['success'] == True:
+        try:
+            ar_log_monitor.start(timeout=60, callback=wait_message_line)
+        except TimeoutError as err:
+            raise AssertionError("AR message tooks too much!")
+
+        # Checking shutdown message in ossec logs
+        try:
+            ossec_log_monitor.start(timeout=60, callback=wait_shutdown_message_line)
+        except TimeoutError as err:
+            raise AssertionError("Shutdown message tooks too much!")
+
+        try:
+            ar_log_monitor.start(timeout=60, callback=wait_ended_message_line)
+        except TimeoutError as err:
+            raise AssertionError("Ended message tooks too much!")
+
+    else:
+        try:
+            ar_log_monitor.start(timeout=60, callback=wait_invalid_input_message_line)
+        except TimeoutError as err:
+            raise AssertionError("Invalid input message tooks too much!")
```

Comment:
```suggestion
        except TimeoutError:
            raise AssertionError("Received message took too much!")
```

These lines are repeated over the code. I think it worth creating a function to run them:
```python
def start_log_monitoring(monitor, callback, timeout=60):
    try:
        monitor.start(timeout, callback)
    except TimeoutError as err:
        raise AssertionError("Invalid input message tooks too much!")